### DISCUSSION TITLE
Noise removal beyond echopype

### DISCUSSION THREAD

**Discussion by martin-wegmann** [OSOceanAcoustics/echopype#1090] [General]
Dear Echopype community,

I am an environmental data scientist working together with fish biologists and so far I used echopype a great deal to convert .raw data into MVBS NetCDF data. Your library is amazing and so intuitive and the biologist who I support are very happy about the post-processed data!!

I was wondering, if this community has experience in removing recurring artificial signals from echo sounder data, which would probably be more like image cleaning, but I thought I am gonna ask you for inspiration first. 

Attached you find two time periods where other instruments in the water (profiler that go up and down) introduce artificial, recurring noise in MVBS data created by the echopype library. Visually clearly detectable, but removing that noise on a code level is tricky, since there are slight variations in reflection power, frequency etc.

Does anyone of you have experience with an issue like this? 
Or do you have ideas in which direction to go?

I would appreciate any input a great deal,
cheers,
Martin

![case_study_B](https://github.com/OSOceanAcoustics/echopype/assets/91071414/da0438d4-33db-43fd-880a-5deec18d8895)
![case_study_A](https://github.com/OSOceanAcoustics/echopype/assets/91071414/d35ed6d2-b54e-491a-ab4c-040cf567f7b2)

**Comment by leewujung** [comment]
@martin-wegmann : Great you've had good experience with echopype! What you showed is a very common type of impulsive noise that echosounder has, often due to interference from other instruments as you said in this case. It is in our plan to add functions to remove such impulsive noise (#727) and other common type of noise, but we haven't gotten around to implement it. You could see some of it in [this paper](https://academic.oup.com/icesjms/article/72/8/2482/2459093) (the "IN" refers to “spike” or impulsive noise like in your case) and there are many more recent papers too. There are for sure other ways to remove such noise, but comparing the ping-to-ping magnitude is a common and simple approach. One thing important is to do the noise removal _before_ you do `compute_MVBS`, so that the noise does not propagate down. If you are interested in contributing something, let us know and we can facilitate the process!

**Reply by martin-wegmann** [reply]
Thank you very much for your feedback, I appreciate it!
